modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -2111,3 +2111,17 @@ def call_from_frame(df):
             return pd.MultiIndex.from_frame(df, sortorder)
 
     eval_general(modin_df, pandas_df, call_from_frame, comparator=assert_index_equal)
+
+
+def test__getitem_bool_single_row_dataframe():
```

Comment:
I think this test came from the original example in #4845. It will be clearer if we can trim the reproducer down as much as possible. I don't think we need the `contains` or `astype`. Can we do something like 

```python
df = pd.DataFrame([1])
result = df[pd.Series(True)]
```

and compare to doing the same for pandas? We should also check that the smaller test case still reproduces the bug.

Reply:
I was just thinking about how we could trim down the test. I think your suggestion is better.

Reply:
We could probably do:
```python
modin_df, pandas_df = create_test_dfs([1])
eval_general(modin_df, pandas_df, lambda df: df[pd.Series(True)])
```

Reply:
I'm not sure why, but when I run this reproduce in ipython, it reproduces the same error. However, when I write it as a test, the test always passes.

When I use the original reproducer, the test passes/fails as expected.

Reply:
You should not be running `eval_general()` like that, as pandas won't accept Modin Series (yours `pd.Series()`) here.

Reply:
I wonder if we can do smth like 
```python
eval_general(pd, pandas, lambda lib: lib.DataFrame([1])[lib.Series([True])])
```